cmd/glasskube/cmd/repo_delete.go

```diff
@@ -0,0 +1,96 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/glasskube/glasskube/api/v1alpha1"
+	"github.com/glasskube/glasskube/internal/cliutils"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var repoDeleteCmd = &cobra.Command{
+	Use:    "delete [repositoryName]",
+	Short:  "Delete a repository",
+	Args:   cobra.ExactArgs(1),
+	PreRun: cliutils.SetupClientContext(true, &rootCmdOptions.SkipUpdateCheck),
+	Run: func(cmd *cobra.Command, args []string) {
+		repositoryName := args[0]
+		ctx := cmd.Context()
+		deleteRepository(ctx, repositoryName)
+	},
+}
+
+func deleteRepository(ctx context.Context, repoName string) {
+	client := cliutils.PackageClient(ctx)
+
+	var repos v1alpha1.PackageRepositoryList
+	if err := client.PackageRepositories().GetAll(ctx, &repos); err != nil {
+		fmt.Fprintf(os.Stderr, "❌ error listing package repository: %v\n", err)
+		cliutils.ExitWithError()
+	}
+
+	var targetRepo *v1alpha1.PackageRepository
+	for i := range repos.Items {
+		if repos.Items[i].Name == repoName {
+			targetRepo = &repos.Items[i]
+			break
+		}
+	}
+
+	if targetRepo == nil {
+		fmt.Fprintf(os.Stderr, "❌ repository %s not found\n", repoName)
+		cliutils.ExitWithError()
+	}
```

Comment:
you can simply use `client.PackageRepositories().Get(ctx, repoName, &targetRepo)`, where `targetRepo` is a `v1alpha.PackageRepository`. That way only the one repository is actually fetched from the server, and you don't need to filter manually. Then you also don't need to check for `targetRepo == nil`, because if that were the case, `err != nil` and the error message would be printed.